### DISCUSSION TITLE
segmentation fault (core dumped)

### DISCUSSION THREAD

**Discussion by ghost** [mkckr0/audio-share#20] [Q&A]
Hello, i am getting a error when i run `as-cmd -b 192.168.1.2`.
output:
`[2023-12-02 21:00:07.410] [info] pipewire header_version: 0.3.48, library_version: 1.0.0`
`zsh: segmentation fault (core dumped)  as-cmd -b 192.168.1.2`

What am i missing?

**Comment by mkckr0** [comment]
Your pipewire library version is `1.0.0`, which is latest, but may not be compatible with Audio Share. I suggest you to install a pipewire client library with a version `0.3.xx`.

**Reply by ghost** [reply]
I downgraded the version to 0.3.85, and the error is different now.
input:
`as-cmd -b 192.168.1.2`
output:
`[2023-12-03 12:39:57.316] [info] pipewire header_version: 0.3.48, library_version: 0.3.85`
`endpoint_id is empty`
Any idea?

**Comment by mkckr0** [comment]
Your PipeWire is not setup properly. You should google the instruction of how to install pipewire on your Linux distribution.

**Reply by ghost** [reply]
Sorry man i'm not so good in audio stuff of linux. I use pulseaudio for sound, so should i replace it with pipewire-pulse or something else?

Btw pipewire service is running fine
input:
`systemctl --user status pipewire.service`
output:
```
● pipewire.service - PipeWire Multimedia Service
     Loaded: loaded (/usr/lib/systemd/user/pipewire.service; disabled; preset: enabled)
     Active: active (running) since Sun 2023-12-03 21:02:45 IST; 9min ago
TriggeredBy: ● pipewire.socket
   Main PID: 7592 (pipewire)
      Tasks: 3 (limit: 4307)
     Memory: 1.6M
        CPU: 19ms
     CGroup: /user.slice/user-1000.slice/user@1000.service/session.slice/pipewire.service
             └─7592 /usr/bin/pipewire

Dec 03 21:02:45 arch systemd[481]: Started PipeWire Multimedia Service.
```

**Reply by mkckr0** [reply]
What's your Linux distro?

**Reply by ghost** [reply]
Arch

**Reply by mkckr0** [reply]
You should follow the instruction of https://wiki.archlinux.org/title/PipeWire#PulseAudio_clients.

Find any test.wav file from Internet, and run `pw-play -v test.wav` to play this file to see whether pipewire works fine.

**Reply by ghost** [reply]
As mentioned in wiki, i replaced pulseaudio to pipewire-pulse and now its working fine. Thanks for your support.
And also there is 2-3 second delay in sound, can i do something about it?

**Reply by mkckr0** [reply]
You can use USB tethering. If it still exists, you may have the same problem as #16. It's a known bug.

**Reply by mkckr0** [reply]
You may have same problem as #31.
